src/_gettsim/policy_environment.py

```diff
@@ -1,59 +1,166 @@
+from __future__ import annotations
+
 import copy
 import datetime
 import operator
-from collections.abc import Callable
 from functools import reduce
+from typing import TYPE_CHECKING, Any
 
 import numpy
 import pandas as pd
 import yaml
 
 import _gettsim.functions  # Execute all decorators # noqa: F401
 from _gettsim.config import INTERNAL_PARAMS_GROUPS, RESOURCE_DIR
-from _gettsim.functions_loader import load_internal_functions
+from _gettsim.functions_loader_new import load_functions_for_date
 from _gettsim.piecewise_functions import (
     check_thresholds,
     get_piecewise_parameters,
     piecewise_polynomial,
 )
+from _gettsim.policy_function import PolicyFunction
 
+if TYPE_CHECKING:
+    from collections.abc import Callable
 
-def set_up_policy_environment(date):
-    """Set up the policy environment for a particular date.
 
-    Parameters
-    ----------
-    date : int, str, datetime.date
-        The date for which the policy system is set up.
+class PolicyEnvironment:
+    """
+    A container for policy functions and parameters. You should usually use the factory
```

Comment:
```suggestion
    A container for policy functions and parameters. 
    
    Almost always, instances are created with :PolicyEnvironment.for_date()`.
```

Reply:
I do appreciate the deeper explanation, but I also think it will be confusing to most potential readers.